sumpy/tools.py

```diff
@@ -579,6 +579,8 @@ class ScalingAssignmentTag(Tag):
 class KernelComputation:
     """Common input processing for kernel computations."""
 
+    default_name: ClassVar[str] = "unknown"
```

Comment:
Abstract property?

Reply:
Yeah, made this and `get_kernel` abstract and updated all the classes in 5d9a1d1.